src/client/sync/tcp.rs

```diff
@@ -15,9 +13,11 @@ pub fn connect(socket_addr: SocketAddr) -> Result<Context> {
 /// gateway that is forwarding messages to/from the corresponding unit identified
 /// by the slave parameter.
 pub fn connect_slave(socket_addr: SocketAddr, slave: Slave) -> Result<Context> {
-    let mut core = Core::new()?;
-    let handle = core.handle();
-    let async_ctx = core.run(async_connect_slave(&handle, socket_addr, slave))?;
-    let sync_ctx = Context { core, async_ctx };
+    let mut rt = tokio::runtime::Runtime::new().unwrap();
```

Comment:
Why not map the error instead of `unwrap`?

Reply:
fixed